src/nimblepkg/forge_aliases.nim

```diff
@@ -0,0 +1,120 @@
+# Copyright (C) Dominik Picheta. All rights reserved.
+# BSD License. Look at license.txt for more info.
+
+import std/[strutils]
+import common, download
+
+type
+  ForgeKind* = enum
+    fgGitHub
+    fgGitLab
+    fgSourceHut
+    fgCodeberg
+
+  Forge* = ref object
+    case kind*: ForgeKind
+    of fgGitHub:
+      ghUsername*, ghRepo*: string
+    of fgGitLab:
+      glUsername*, glRepo*: string
+    of fgSourceHut:
+      shUsername*, shRepo*: string
+    of fgCodeberg:
+      cbUsername*, cbRepo*: string
```

Comment:
```suggestion
type
  ForgeKind* = enum
    fgGitHub
    fgGitLab
    fgSourceHut
    fgCodeberg

  Forge* = object
    kind*: ForgeKind
    username*, repo*: string
```

There seems to be no advantage to require the user of the `Forge` type to do the case analysis.

Reply:
Yeah, seems like it. I was originally planning to add self-hosted forges into this as well (as @jtbx suggested) but I later realized it'd be counter intuitive. I'll fix this.